kerykeion/secondary_progressions/solar_arc_factory.py

```
@@ -0,0 +1,231 @@
+            A :class:`SolarArcSubjectModel` describing the arc and every
+            directed point + directed-to-natal aspect found.
+        """
+        if natal_subject.sun is None:
+            raise KerykeionException("Natal subject is missing the Sun — cannot compute solar arc.")
+
+        target_utc = _parse_target_utc(target_iso_utc_datetime, target_year)
+
+        progressed = SecondaryProgressionFactory.compute(
+            natal_subject,
+            target_iso_utc_datetime=target_utc.strftime("%Y-%m-%dT%H:%M:%S.000Z") if target_utc else None,
+            target_year=None if target_utc else None,
```

Ruff (0.15.12)

[warning] Useless `if`-`else` condition

(RUF034)